pkg/operator/operator.go

```diff
@@ -265,35 +269,40 @@ func isMachineWebhook(obj interface{}) bool {
 }
 
 func (optr *Operator) worker() {
-	for optr.processNextWorkItem() {
+	var cancel context.CancelFunc
+	processNext := true
+	optr.context, cancel = context.WithCancel(context.Background())
+	for processNext {
+		processNext, cancel = optr.processNextWorkItem(cancel)
```

Comment:
What is the purpose of getting a cancel function here?

Reply:
The current worker gets canceled on a long-running task by the next one, like waiting for rollout, if he arrives earlier then it finishes. Just making sure we account for the changes indicated by the event. 1 worker for one request, but with that, we are giving priority to the state maintenance, rather than a full flow completion. Exposing the `cancel` here provides the next worker an ability to stop his predecessor and take its place.